--- before
+++ after
@@ -1,6 +1,12 @@
 see the files COPYING3 and COPYING.RUNTIME respectively.  If not, see
 <http://www.gnu.org/licenses/>.  */
+
+/* Systems using musl libc should use this header and make sure
+   OPTION_MUSL is defined correctly before using the TYPE macros. */
+#ifndef OPTION_MUSL
+#define OPTION_MUSL 0
+#endif
 
 #define SIG_ATOMIC_TYPE "int"
 
 #define INT8_TYPE "signed char"
